pkg/streamingpromql/compat/fallback_engine.go

```diff
@@ -27,6 +27,8 @@ type EngineWithFallback struct {
 	logger log.Logger
 }
 
+var errFallbackForcedByHTTPHeader = NotSupportedError{"fallback forced by HTTP header"}
```

Comment:
(nit)
I don't know if I like using `NotSupportedError` here. It's not that the query is necessarily unsupported, but that we aren't going to use the query engine. Maybe we need an error class "MQEError" or something that NotSupportError and Fallback use?

Similarly renaming `e.unsupportedQueries` to FallbackQueries or similar.

Reply:
Good point, I've dropped the awkward use of `NotSupportedError` in 28c4bc2.

I've kept `unsupportedQueries` as-is for now - if it annoys us in the future, we can change it.